lib/client.js

```diff
@@ -537,21 +638,34 @@ MqttClient.prototype.subscribe = function () {
  * unsubscribe - unsubscribe from topic(s)
  *
  * @param {String, Array} topic - topics to unsubscribe from
+ * @param {Object} [opts] - optional subscription options, includes:
+ *    {Object} properties - properties of unsubscribe packet
  * @param {Function} [callback] - callback fired on unsuback
  * @returns {MqttClient} this - for chaining
  * @api public
  * @example client.unsubscribe('topic');
  * @example client.unsubscribe('topic', console.log);
  */
-MqttClient.prototype.unsubscribe = function (topic, callback) {
+MqttClient.prototype.unsubscribe = function () {
   var packet = {
     cmd: 'unsubscribe',
     qos: 1,
     messageId: this._nextId()
   }
   var that = this
+  var args = Array.prototype.slice.call(arguments)
```

Comment:
This is better implemented with:

```js
var args = new Array(arguments.length)
for (var i = 0; i < arguments.length; i++) {
  args[i] = arguments[i]
}
```

because using `Array.prototype.slice` is extremely unperformant in older Node.js versions.

Reply:
implemented